addons/project/views/project_views.xml

```diff
@@ -1056,6 +1071,12 @@
                             <field name="project_id" attrs="{'invisible': [('parent_id', '!=', False)]}" domain="[('active', '=', True), ('company_id', '=', company_id)]"
                                 placeholder="Private" class="o_project_task_project_field oe_edit_only"/>
                             <field name="display_project_id" string="Project" attrs="{'invisible': [('parent_id', '=', False)]}" domain="[('active', '=', True), ('company_id', '=', company_id)]"/>
+                            <field name="milestone_id"
+                                placeholder="e.g. Product Launch"
+                                context="{'default_project_id': project_id if not parent_id or not display_project_id else display_project_id}"
+                                attrs="{'invisible': ['|', ('project_id', '=', False), ('allow_milestones', '=', False)]}"
+                                groups="project.group_project_milestone"
+                            />
```

Comment:
```suggestion
                            <field name="milestone_id"
                                placeholder="e.g. Product Launch"
                                context="{'default_project_id': project_id if not parent_id or not display_project_id else display_project_id}"
                                attrs="{'invisible': ['|', ('project_id', '=', False), ('allow_milestones', '=', False)]}"
                            />
```
I think the group is unnecessary here as it is encompassed by `allow_milestones`